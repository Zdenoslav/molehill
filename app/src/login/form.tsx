import React from 'react'
import { compose } from 'redux';
import { Form, InjectedFormProps, reduxForm } from 'redux-form'

import Button from 'app/components/button';
import FormField from 'app/components/form-elements/form-field';
import FormInput from 'app/components/form-elements/form-input';

interface IFormData {
  email: string;
  password: string;
}

type Props = InjectedFormProps<IFormData>;

const LoginForm: React.SFC<Props> = props => {
  const { handleSubmit } = props

  return (
    <Form onSubmit={handleSubmit}>
        <FormField name="email" component={FormInput} placeholder="Email address" />
        <FormField name="password" component={FormInput} type="password" placeholder="Password" />
        <Button type="submit" appearance="submit" fullWidth={true} />
    </Form>
  );
}

export default compose(
  reduxForm({
    form: 'LOGIN_FORM',
  }),
)(LoginForm);