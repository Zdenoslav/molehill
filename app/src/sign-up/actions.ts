import {createActions} from 'app/redux/create-actions';

const domain = 'SIGN_UP';

export const {
  action: signUp,
  success: signUpSuccess,
} = createActions(
  `${domain}`,
  (formData) => ({formData}),
);
