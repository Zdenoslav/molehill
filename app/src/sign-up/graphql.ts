import gql from 'graphql-tag';

export const signUpMutation = gql`
  mutation createUser($user: SignUpInput!) {
    createUser(user: $user) {
      id
      username
      email
    }
  }
`;
