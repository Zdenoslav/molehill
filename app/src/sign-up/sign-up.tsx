import React from 'react';
import { Link } from 'react-router-dom';

import ScreenCenter from 'app/components/screen-center';
import { Body, Headline } from 'app/components/text';

import Form, {ISignUpFormData} from './form';

const SignUp = () => {

  const handleSubmit = (values: ISignUpFormData) => {
    // tslint:disable-next-line:no-console
    console.log(values);
  };

  return (
    <ScreenCenter>
      <Headline textAlign="center">Molehill</Headline>
      <Body textAlign="center">Sign up to see what people around you are up to</Body>
      <Form onSubmit={handleSubmit} />
      <Body textAlign="center">Do you have an account already? <Link to="/login">Log in</Link></Body>
    </ScreenCenter>
  );
};

export default SignUp;
