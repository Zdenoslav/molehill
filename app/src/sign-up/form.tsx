import React from 'react';
import { compose } from 'redux';
import { Form, InjectedFormProps, reduxForm } from 'redux-form';

import Button from 'app/components/button';
import FormField from 'app/components/form-elements/form-field';
import FormInput from 'app/components/form-elements/form-input';

import { email, maxLength, minLength } from 'app/helpers/form-validation';
import { SignUpInput } from 'app/generated/graphql';

const minLength3 = minLength(3);
const maxLength24 = maxLength(24);

export interface ISignUpFormData extends SignUpInput {
  passwordRepeat: string;
}

type Props = InjectedFormProps<ISignUpFormData>;

const SignUpForm: React.SFC<Props> = props => {
  const { handleSubmit } = props;

  return (
    <Form onSubmit={handleSubmit}>
        <FormField
          required
          validate={email}
          type="email"
          name="email"
          component={FormInput}
          placeholder="Email address"
        />
        <FormField
          required
          validate={[minLength3, maxLength24]}
          name="username"
          component={FormInput}
          placeholder="Username"
        />
        <FormField required name="password" component={FormInput} type="password" placeholder="Password" />
        <FormField required name="passwordRepeat" component={FormInput} type="password" placeholder="Repeat password" />
        <Button text="Sign up" type="submit" appearance="submit" fullWidth />
    </Form>
  );
};

const validate = (values: ISignUpFormData) => {
  const errors: Partial<ISignUpFormData> = {};

  if (values.password !== values.passwordRepeat) {
    errors.passwordRepeat = `Those passwords didn't match. Try again.`;
  }
  return errors;
};

export default compose(
  reduxForm({
    form: 'SIGNUP_FORM',
    validate,
  }),
)(SignUpForm);
