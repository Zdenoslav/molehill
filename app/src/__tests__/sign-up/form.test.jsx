import React from 'react';
import { shallow } from 'enzyme';

import {SignUpForm, validate} from 'app/sign-up/form';

describe('SignUpForm', () => {

  it('should render SignUpForm', () => {
    const wrapper = shallow(<SignUpForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should return empty errors if form data is valid', () => {
    const formData = {
      password: 'secret',
      passwordRepeat: 'secret',
    };
    const errors = validate(formData);

    expect(errors).toEqual({});
  });

  it('should return errors if form data is invalid', () => {
    const formData = {
      password: 'secret',
      passwordRepeat: 'different pass',
    };
    const errors = validate(formData);
    const expectedErrors = {
      passwordRepeat: `Those passwords didn't match. Try again.`,
    };

    expect(expectedErrors).toEqual(errors);
  });

});
