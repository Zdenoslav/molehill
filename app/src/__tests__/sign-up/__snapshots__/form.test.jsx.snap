// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm should render SignUpForm 1`] = `
<Form>
  <FormField
    component={[Function]}
    name="email"
    placeholder="Email address"
    required={true}
    type="email"
    validate={[Function]}
  />
  <FormField
    component={[Function]}
    name="username"
    placeholder="Username"
    required={true}
    validate={
      Array [
        [Function],
        [Function],
      ]
    }
  />
  <FormField
    component={[Function]}
    name="password"
    placeholder="Password"
    required={true}
    type="password"
  />
  <FormField
    component={[Function]}
    name="passwordRepeat"
    placeholder="Repeat password"
    required={true}
    type="password"
  />
  <Button
    appearance="submit"
    fullWidth={true}
    text="Sign up"
    type="submit"
  />
</Form>
`;
